src/Stratis.Bitcoin.Features.SmartContracts/SmartContractPosRuleRegistration.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Collections.Generic;
+using Stratis.Bitcoin.Consensus.Rules;
+using Stratis.Bitcoin.Features.Consensus.Rules.CommonRules;
+using Stratis.Bitcoin.Features.SmartContracts.Consensus.Rules;
+
+namespace Stratis.Bitcoin.Features.SmartContracts
+{
+    public class SmartContractPosRuleRegistration : IRuleRegistration
+    {
+        public IEnumerable<ConsensusRule> GetRules()
+        {
+            return new List<ConsensusRule>
+                {
+                    new TemporarySetChainHeader(),
+
+                    // == Header ==
+                    new HeaderTimeChecksRule(),
+                    new HeaderTimeChecksPosRule(),
+                    new StratisBigFixPosFutureDriftRule(),
+                    new CheckDifficultyPosRule(),
+                    new StratisHeaderVersionRule(),
+
+                    // == Integrity ==
+                    new BlockMerkleRootRule(),
+                    new SmartContractPosBlockSignatureRule(),
+
+                    // == Partial ==
+                    new SetActivationDeploymentsRule(),
+                    new CheckDifficultykHybridRule(),
+                    new PosTimeMaskRule(),
+
+                    // rules that are inside the method CheckBlockHeader
```

Comment:
blank block ?

Reply:
Jumping on this - do we need anything here Francois?

Reply:
This was copied from PosRuleRegistration, we can consolidate once FN has completed the "big" consensus refactor 👍